internal/broker/broker.go

```diff
@@ -191,6 +181,23 @@ func (b *Broker) NewSession(username, lang, mode string) (sessionID, encryptionK
 	return sessionID, base64.StdEncoding.EncodeToString(pubASN1), nil
 }
 
+func (b *Broker) connectToProvider(ctx context.Context) (authCfg authConfig, err error) {
+	ctx, cancel := context.WithTimeout(ctx, maxRequestTime)
+	defer cancel()
+
+	authCfg.provider, err = oidc.NewProvider(ctx, b.issuerURL)
```

Comment:
I don’t like that we are augmenting the object little by little and it’s not really the pattern that we have.

In general, I think we should assign to new variables, like provider and oauth and `return authConfig{provider: provider, oauth: oauth}, nil` (probably in multiple lines like the other constructors.